### ISSUE TITLE
How should the `server` option be used?

### ISSUE THREAD

**Issue by christianbundy** [ssbc/multiserver#33]
It looks like we're [having problems](https://github.com/ssbc/scuttlebot/issues/577) that seem to be [resolved by `server: true`](https://github.com/ssbc/patchbay/blob/e6ed52fd655a6ba06fe6d3b2e9a6646b93dcf903/config.js#L33). From skimming the source, I can see `opts.server` used in three places:

```console
$ git grep opts.server
plugins/onion.js:      if(!opts.server) return
plugins/onion.js:      if(opts && !opts.server) return null
plugins/unix-socket.js:    server: !opts.server ? null : function (onConnection) {
plugins/unix-socket.js:      if(opts && !opts.server) return null
plugins/ws.js:  var secure = opts.server && !!opts.server.key
plugins/ws.js:      var server = opts.server || http.createServer(opts.handler)
plugins/ws.js:      if(!opts.server) {
plugins/ws.js:      if(opts.server)
plugins/ws.js:        port = opts.server.address().port
```

- **Tor:** when `opts.server` is falsy, it looks like the connection is abandoned and we instead just `return`.
 - **Unix socket:** Same as above, but instead of a function that immediately returns the function is being replaced by `null`. Is there a reason for the difference in these two implementations?
- **Websocket:** It looks like `opts.server` is an object with two expected properties: `key` and `address`, and the latter is a function that returns an object with the property `port`. The full usage is `opts.server.address().port`, which makes me think there are some moving parts here I don't understand.

@dominictarr Could you add some info to help me better understand what we should be doing with this option? Apologies if this has already been recorded somewhere, so far I've checked #32, ssb-config, and scuttlebot but didn't see any notes about this change. Appreciate your help!

cc: @arj03 @cryptix

**Comment by arj03** [comment]
@christianbundy Great grepping. I'll try to clear this up as its really affecting everyone upgrading.

**Comment by christianbundy** [comment]
@arj03 Forking this conversation to the issue rather than your PR:

> You did a nice analysis in #33 and from that only ws is left, which was the original meaning of server.

Could you explain this another way? Sorry, I'm just not getting the semantics behind `server`. I imagine it doesn't mean "this is a websocket server", because otherwise it'd be `ws: true` or something, so I'm trying to come up with a model of which interfaces use the property (and what the value might be).

**Comment by arj03** [comment]
@christianbundy I had to go hunting as I have never used the WS plugin properly before, but at least I learned something that I have tried to document [here](https://github.com/ssbc/ssb-config/pull/42) :)

**Comment by arj03** [comment]
I think this should be closed now by multiserver 3.3.0 and latest ssb-config
